### ISSUE TITLE
Links in the documentation should not specify docs.rs path

### ISSUE THREAD

**Issue by brycx** [brycx/orion#91]
As @vlmutolo mentioned in [this issue](https://github.com/brycx/orion/pull/86#discussion_r319771859), the current hyperlinks in the documentation all link to docs.rs.

These should be changed to follow the [Rust API guidelines](https://rust-lang-nursery.github.io/api-guidelines/documentation.html#prose-contains-hyperlinks-to-relevant-things-c-link) and not specifically link to docs.rs if the item is part of orion. This will make development easier, since the links will automatically point to locally built docs, instead of docs.rs.

The new format will probably look as follows (example taken from `orion::aead`, not tested to work with docs.rs):
```
//! [`seal`]: fn.seal.html
//! [`open`]: fn.open.html
//! [`POLY1305_OUTSIZE`]: ../hazardous/mac/poly1305/constant.POLY1305_OUTSIZE.html
//! [`XCHACHA_NONCESIZE`]: ../hazardous/stream/xchacha20/constant.XCHACHA_NONCESIZE.html
//! [`SecretKey::default()`]: struct.SecretKey.html
```

**Comment by vlmutolo** [comment]
I can take this one on. I think it’s just `grep`ing for `http` in the source.

**Comment by brycx** [comment]
I just found this very nice tool that we can use to check that all the links are valid: https://github.com/deadlinks/cargo-deadlinks

Maybe we should even consider adding this tool to the CI documentation build job.

**Comment by brycx** [comment]
Fixed in #100.
